client/src/component/AddVideo.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect } from "react";
+
+function Addvideo() {
+  const [videos, setVideos] = useState([]);
+  const [newVideo, setNewVideo] = useState({ title: "", url: "" });
+  const [idCounter, setIdCounter] = useState(1);
+
+  useEffect(() => {
+    fetchVideos();
+  }, []);
+
+  const fetchVideos = async () => {
+    try {
+      const response = await fetch(
+        "https://node-server-full-stack.onrender.com/videos"
+      );
+      if (response.ok) {
+        // const data = await response.json();
+        // setVideos(data);
+      } else {
+        console.error("Failed to fetch videos:", response.statusText);
+      }
+    } catch (error) {
+      console.error("Error fetching videos:", error);
+    }
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setNewVideo({ ...newVideo, [name]: value });
+  };
+
+  const getYouTubeVideoId = (url) => {
+    const videoIdMatch = url.match(/(?:\/|v=)([A-Za-z0-9_-]{11})(?=&|$)/);
+    if (videoIdMatch) {
+      return videoIdMatch[1];
+    }
+    return null;
+  };
+
+  const addVideo = async () => {
+    if (newVideo.title && newVideo.url) {
+      const videoId = getYouTubeVideoId(newVideo.url);
+
+      if (videoId) {
+        const videoToAdd = {
+          ...newVideo,
+          rating: 0,
+          id: idCounter,
```

Comment:
Ordinarily we'd expect the server to choose an ID, rather than the client, because a client can't know what IDs the server already knows about.

It looks like the server ignores this field completely, so it seems strange to set it. I think you can stop setting it, and also remove the `idCounter` state completely?